src/collections/fifo_mt.c

```diff
@@ -14,44 +14,33 @@
 
 #include "zenoh-pico/collections/fifo_mt.h"
 
+#include "zenoh-pico/protocol/codec/core.h"
 #include "zenoh-pico/protocol/core.h"
 #include "zenoh-pico/utils/logging.h"
 
 /*-------- Fifo Buffer Multithreaded --------*/
 int8_t _z_fifo_mt_init(_z_fifo_mt_t *fifo, size_t capacity) {
-    int8_t res = _z_fifo_init(&fifo->_fifo, capacity);
-    if (res) {
-        return res;
-    }
+    _Z_RETURN_IF_ERR(_z_fifo_init(&fifo->_fifo, capacity))
 
 #if Z_FEATURE_MULTI_THREAD == 1
-    res = zp_mutex_init(&fifo->_mutex);
-    if (res) {
-        return res;
-    }
-    res = zp_condvar_init(&fifo->_cv_not_full);
-    if (res) {
-        return res;
-    }
-    res = zp_condvar_init(&fifo->_cv_not_empty);
-    if (res) {
-        return res;
-    }
+    _Z_RETURN_IF_ERR(zp_mutex_init(&fifo->_mutex))
```

Comment:
Still needs to replace the deprecrated systems calls: https://github.com/eclipse-zenoh/zenoh-pico/blob/main/src/deprecated/platform.c

Reply:
Just realized this changes won't be in this branch until #394 is merged. So we either accept as is and modify later  or we wait till #394 is merged and rebase on that. Opinions @Mallets?

Reply:
I suggest to ago ahed with this PR and we then update #394. @jean-roland what do you think?